hardware/OpenCustomizationSelector/src/com/sonymobile/customizationselector/Parser/ModemConfParser.java

```diff
@@ -75,6 +75,9 @@ private static String parseModemFileName(File modemFile) {
                 CSLog.e(TAG, "IOException: while closing reader", e);
             }
         }
+        if (modemVariant == "" || modemVariant == null) {
```

Comment:
`modemVariant` cannot be `null`. And you also miss the early return at https://github.com/Flamefire/android_device_sony_yoshino-common/pull/40/files#diff-3b685ccc21f9b13cb490b5f12ae11083b5f8c70e04e08b806e1924d951c0c1edR53

Especially the latter point suggests that this isn't the right place to put this code. I mean `parseModemFileName` should just do that. The defaulting to another modem should be done higher up. A better place would be `getCarrierBundle` or even at the place of use in `evaluateModem`: Change the parameter value to "ir51_ir92_ims" if it is empty

BTW: Earlier you described this as
> CustomizationSelector was also modified to default to ir51_ir92_ims when IMS Features are enabled and when no customizations are found.

However I don't see an (explicit) check for "when IMS Features are enabled". Could this change use this modem even when IMS is disabled? Or how is this avoided?

Reply:
Okay, yeah, the early return was a significant miss. However, isn't `evaluateModem` executed after `parseModemFileName`?

I also avoided modifying overly abstract functions, so I don't end up breaking something, hence my modification here.

As for "when IMS Features are enabled", the parser is turned off when IMS is turned off. Check `EventReceiver.java`: "IMS disabled, not parsing".

Reply:
> Okay, yeah, the early return was a significant miss. However, isn't `evaluateModem` executed after `parseModemFileName`?

Yep and that's exactly the point: The result of `parseModemFileName` (through any return including the early one) is ultimately passed to `evaluateModem` which has the job of determining the modem to use. So not only do we catch the early return value "automatically" there it also makes more sense to apply the default in a function whose purpose is to evaluate the value/modem-to-use.

> As for "when IMS Features are enabled", the parser is turned off when IMS is turned off. Check `EventReceiver.java`: "IMS disabled, not parsing".

I see. It returns there and doesn't start the `CustomizationSelectorService`

> EDIT: Also, not sure if other swticher functions are turned off as well when IMS is off, and **this** is the real reason why I made my modification here.

If `evaluateModem` is still called we'd need to check that thoroughly if it even makes sense. Or maybe play it safe and check for `(Settings.System.getInt(context.getContentResolver(), "cs_ims", 1) == 0)` before setting the IMS modem as default.